src/peep/peep.c

```diff
@@ -7131,70 +7131,115 @@ uint8 sub_69A60A(rct_peep* peep){
 
 /**
  *
+ *  rct2: 0x0069A997
+ */
+static int sub_69A997(sint16 x, sint16 y, uint8 z, int direction, rct_map_element *dest_map_element, int test_edge) {
+	int eax = x, ebx = direction, ecx = y, edx = z, esi = (int) dest_map_element, edi = 0xFFFF, ebp = test_edge;
+	RCT2_CALLFUNC_X(0x0069A997, &eax, &ebx, &ecx, &edx, &esi, &edi, &ebp);
```

Comment:
I wasn't sure what got used by the other function, so I played it safe. If you know that neither are used, then I'll remove them. 

You also said that dh was a separate variable. Does it have a value I should be initializing it to? As far as I could tell, it was just being assigned `z` from the caller function.